main/test/ca/uwaterloo/flix/library/TestMutSet.flix

```diff
@@ -13,238 +13,370 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 namespace TestMutSet {
 
     /////////////////////////////////////////////////////////////////////////////
     // sameElements                                                            //
     /////////////////////////////////////////////////////////////////////////////
 
     @test
-    def sameElements01(): Bool & Impure =
-        let s = MutSet.empty(): MutSet[Int32];
-        let s2 = MutSet.empty(): MutSet[Int32];
+    def sameElements01(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r): MutSet[Int32, _];
+        let s2 = MutSet.emptyWithRegion(r): MutSet[Int32, _];
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements02(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements02(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements03(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements03(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(2, s2);
 
         not MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements04(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements04(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!(1, s);
         MutSet.add!(2, s);
         MutSet.add!(3, s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!(2, s2);
         MutSet.add!(3, s2);
         MutSet.add!(1, s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def sameElements05(): Bool & Impure =
-        let s = MutSet.empty();
+    def sameElements05(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!("a", s);
         MutSet.add!("b", s);
         MutSet.add!("c", s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!("c", s2);
         MutSet.add!("a", s2);
         MutSet.add!("b", s2);
 
         MutSet.sameElements(s, s2)
+    }
 
     @test
-    def setSameElements06(): Bool & Impure =
-        let s = MutSet.empty();
+    def setSameElements06(): Bool = region r {
+        let s = MutSet.emptyWithRegion(r);
         MutSet.add!("a", s);
         MutSet.add!("b", s);
         MutSet.add!("c", s);
 
-        let s2 = MutSet.empty();
+        let s2 = MutSet.emptyWithRegion(r);
         MutSet.add!("c", s2);
         MutSet.add!("a", s2);
 
         not MutSet.sameElements(s, s2)
+    }
 
```

Comment:
Could I ask you to add some basic tests that uses add, allAll etc? Nothing fancy. 

Just pick like 4 common operations and add 3 tests for each. Thanks!

(This is not really to test MutSet but more to test the region stuff indirectly).

Reply:
I just want a few more tests of core functionality.

Reply:
Maybe something that uses multiple regions.